### PR TITLE
chore(Makefile): add `setup-pre-commit` target to init git hook

### PR DESCRIPTION
**Pull request by levisyin**
### Description

To be honest, we often forget to run `make lint` or check the result of `make lint` before we push the commits

This pr adds a git `pre-commit` hook for linting new changes code, this will prevent contributors from pushing their commits without linting, and reduce the times of re-run ci.

### How to use

After you clone the repository, just simply run `make setup-pre-commit`

image

### Screenshots

image

### CODE REVIEW

**Review by mzabaluev — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -240,6 +240,13 @@ clean_certs:
 ###                  Formatting, linting, and vetting                       ###
 ###############################################################################
 
+#? setup-pre-commit: Set pre-commit git hook
+setup-pre-commit:
```
Comment (melekes) [inline Makefile:244]: There's `gofumpt` pre-commit hook. Can we merge these two?

File: scripts/hooks/pre-commit.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# lint modified go files
+golangci-lint run --fix --new -c .golangci.yml
```
Comment (melekes) [inline scripts/hooks/pre-commit.sh:4]: This can be too slow.. Can we only enable `fast` linters?

File: Makefile (hunk 1 above)
Comment (levisyin) [reply Makefile:244]: Okay, I will merge them

File: scripts/hooks/pre-commit.sh (hunk 2 above)
Comment (levisyin) [reply scripts/hooks/pre-commit.sh:4]: Confusing here, did you mean that we should only lint the changed go files, not all *.go files?

Or did you mean that we should remove `-c .golangci.yml`?

Comment (melekes) [reply scripts/hooks/pre-commit.sh:4]:
```
--fast                           Run only fast linters from enabled linters set (first run won't be fast)
```

**Comment by levisyin** [comment]
@melekes Hi sir, suggestion changes completed

File: Makefile [hunk 3]
```diff
@@ -258,11 +258,12 @@ vulncheck:
...
-#? gofumpt-pre-commit: Create gofumpt pre-commit hook
-gofumpt-pre-commit:
-	@echo "--> Creating gofumpt pre-commit hook"
-	@mkdir -p .git/hooks
-	@cp scripts/gofumpt.sh .git/hooks/pre-commit
+#? setup-pre-commit: Create git pre-commit hook for gofumpt, golangci-lint
+setup-pre-commit:
+	@cp .git/hooks/pre-commit .git/hooks/pre-commit.bak 2>/dev/null || true
+	@echo "Installing pre-commit hook..."
+	@ln -sf ../../scripts/hooks/pre-commit.sh .git/hooks/pre-commit
```
Comment (melekes) [inline Makefile:265]:
```suggestion
	@ln -sf scripts/hooks/pre-commit.sh .git/hooks/pre-commit
```

we're executing this from root dir, no?

File: scripts/hooks/pre-commit.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+# Lint modified go files
+function lint_modified_go_files() {
+    echo "Running golangci-lint on staged Go files:"
+    golangci-lint run --fix --new --fast -c .golangci.yml
+}
+
+format_modified_go_files
+lint_modified_go_files
```
Comment (melekes) [inline scripts/hooks/pre-commit.sh:31]: please add a new line

File: Makefile (hunk 3 above)
Comment (levisyin) [reply Makefile:265]: Yes, we are in root dir. But we should still use `../../scripts/hooks/pre-commit.sh`. I have tested this, `scripts/hooks/pre-commit` couldn't work well.

I think the path is refer to the target to be linked, not the dir that command execute

**Pull request merged**